### ISSUE TITLE
anchor-is-valid

### ISSUE THREAD

**Issue by crobinson42** [evcohen/eslint-plugin-jsx-a11y#520]
I use `<a href="#showall=true">show</a>` to set a url hash, which is valid use of an anchor tag then I use `<a href="#">clear</a>` to reset the hash values, which is valid use of an anchor tag.

I'm still receiving this warning, which I believe is incorrectly being displayed:

```
The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
```

**Comment by ljharb** [comment]
That might depend on your definition of "valid" - storing state in the URL hash is widely considered to be a bad idea, and to "break the web" (http://isolani.co.uk/blog/javascript/BreakingTheWebWithHashBangs https://www.wired.com/2011/02/gawker-learns-the-hard-way-why-hash-bang-urls-are-evil/ etc).

Can you elaborate on the purpose of setting a URL hash?

**Comment by ljharb** [comment]
Typically that's done with pushState and query strings, for SEO as well as many other reasons.

**Comment by crobinson42** [comment]
Ok, then the rational question here is whether this `<a href="#showall=true">` type of href value is against W3 spec use or not.

https://www.w3.org/MarkUp/1995-archive/Elements/A.html

If it is, then I should close this issue. If it's not, should jsx-a11y be warning on use of a hash in the `href` attr?

**Comment by ljharb** [comment]
The definition of the rule seems like it [explicitly disallows](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/src/rules/anchor-is-valid.js#L96-L101) literal `#` - see #224 and https://github.com/evcohen/eslint-plugin-jsx-a11y/pull/219#discussion_r113042485 specifically.
